features/support/api_helpers.rb

```diff
@@ -112,14 +112,42 @@ def register_interest_in_framework(framework_slug, supplier_id)
   end
 end
 
-def submit_supplier_declaration(framework_slug, supplier_id, declaration)
+def submit_supplier_declaration_and_sign_agreement(framework_slug, supplier_id, declaration)
   path = "/suppliers/#{supplier_id}/frameworks/#{framework_slug}/declaration"
   response = call_api(:put, path, payload: {
     declaration: declaration,
     updated_by: "functional tests",
   })
   [200, 201].should include(response.code), _error(response, "Failed to submit supplier declaration #{framework_slug} #{supplier_id}")
   JSON.parse(response.body)['declaration']
+  update_framework_agreement_status(framework_slug, supplier_id, true)
```

Comment:
This "update_supplier_on_framework" (as it actually now is) also feels like a separate step, something like "And the supplier has been awarded onto the framework", which is a thing that happens after they submit their declaration.

I'm not sure it's 100% necessary to set the `on_framework` flag at all though, because I don't think our frontend apps use it - it's mainly used in scripts.  But I could be wrong about that.